pkg/controller/operatingsystemconfig/oscommon/actuator/actuator_reconcile.go

```diff
@@ -0,0 +1,109 @@
+
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actuator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener-extensions/pkg/controller/operatingsystemconfig/oscommon/cloudinit"
+	"github.com/gardener/gardener-extensions/pkg/controller/operatingsystemconfig/oscommon/generator"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (a *Actuator) Reconcile(ctx context.Context, config *extensionsv1alpha1.OperatingSystemConfig) ([]byte, *string, []string, error) {
+	cloudConfig, cmd, err := a.CloudConfigFromOperatingSystemConfig(ctx, config)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("Could not generate cloud config: %v", err)
+	}
+
+	return []byte(cloudConfig), cmd, OperatingSystemConfigUnitNames(config), nil
+}
+
+func (a *Actuator) CloudConfigFromOperatingSystemConfig(ctx context.Context, config *extensionsv1alpha1.OperatingSystemConfig) ([]byte, *string, error) {
+	files := make([]*generator.File, 0, len(config.Spec.Files))
+	for _, file := range config.Spec.Files {
+		data, err := a.DataForFileContent(ctx, config.Namespace, &file.Content)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		files = append(files, &generator.File{Path: file.Path, Content: data, Permissions: file.Permissions})
+	}
+
+	units := make([]*generator.Unit, 0, len(config.Spec.Units))
+	for _, unit := range config.Spec.Units {
+		var content []byte
+		if unit.Content != nil {
+			content = []byte(*unit.Content)
+		}
+
+		dropIns := make([]*generator.DropIn, 0, len(unit.DropIns))
+		for _, dropIn := range unit.DropIns {
+			dropIns = append(dropIns, &generator.DropIn{Name: dropIn.Name, Content: []byte(dropIn.Content)})
+		}
+		units = append(units, &generator.Unit{Name: unit.Name, Content: content, DropIns: dropIns})
+	}
+
+	return a.generator.Generate(&generator.OperatingSystemConfig{
+		Bootstrap: config.Spec.Purpose == extensionsv1alpha1.OperatingSystemConfigPurposeProvision,
+		Files:     files,
+		Units:     units,
+		Path:	   config.Spec.ReloadConfigFilePath,
+	})
+}
+
+func (a *Actuator) DataForFileContent(ctx context.Context, namespace string, content *extensionsv1alpha1.FileContent) ([]byte, error) {
```

Comment:
Instead of creating a 'common' actuator, can you please rather declare this as a top level function, i.e.
```go
DataForFileContent(ctx context.Context, c client.Client, content *extensionsv1alpha1.FileContent) ([]byte, error)
```
Then, an implementor can just call these functions.

Reply:
Done